### PR TITLE
feat: batch queue, permissions and queueDelete endpoint for annotation_id

### PR DESCRIPTION
**Pull request by sri-kirsh**
## Goal
`/queueDelete` logic to batch the `annotation_id` in a sqs message.

### implementation details
- PR for rolling out the batch sqs, permissions and queueDelete logic.
- notes dynamo deletion is called with `/queueDelete` endpoint as we don't have to batch them
- same pattern as that of list-api, and other deletion services
- batchDelete in a follow up PR -  https://github.com/Pocket/annotations-api/pull/266 (merged) 

ticket: https://getpocket.atlassian.net/browse/INFRA-637

### CODE REVIEW

File: src/server/routes/queueDelete.ts [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    toInt: true,
+  },
+  isPremium: {
+    in: ['body'],
+    errorMessage: 'Must provide valid isPremium (true | false)',
+    isBoolean: true,
+  },
+};
+
+router.post(
```
Comment (sri-kirsh) [inline src/server/routes/queueDelete.ts:48]: router is added to main.ts in https://github.com/Pocket/annotations-api/pull/266

File: lambda/events/handlers/accountDelete.ts [hunk 2]
```diff
@@ -3,8 +3,9 @@ import { config } from '../config';
 import fetch from 'node-fetch';
 
 /**
- * Given an account delete event, queue SQS messages to delete user
- * data in the Annotations subgraph.
+ * Given an account delete event, queue SQS messages to delete chunks of the
+ * user's list and tags from the database. Since we don't want to overload the database,
```
Comment (kschelonka) [inline lambda/events/handlers/accountDelete.ts:7]: This looks like a copy-paste error from list-api

File: src/config/index.ts [hunk 3]
```diff
@@ -12,6 +12,20 @@ export default {
...
+      batchSize: 10,
+      annotationsDeleteQueue: {
+        url:
+          process.env.SQS_BATCH_DELETE_QUEUE_URL ||
+          'http://localhost:4566/queue/pocket-annotations-delete-queue',
+        visibilityTimeout: 300,
+        maxMessages: 1,
+        waitTimeSeconds: 0,
+        defaultPollIntervalSeconds: 300,
+        afterMessagePollIntervalSeconds: 60,
```
Comment (kschelonka) [inline src/config/index.ts:26]: I think we want this to be 30 seconds, per the discussion in slack https://pocket.slack.com/archives/C03E28D1GUD/p1657671668758909

File: src/config/index.ts [hunk 4]
```diff
@@ -49,4 +63,8 @@ export default {
     dbName: process.env.DATABASE || 'readitla_ril-tmp',
     tz: process.env.DATABASE_TZ || 'US/Central',
   },
+  queueDelete: {
+    queryLimit: 500,
+    itemIdChunkSize: 100,
```
Comment (kschelonka) [inline src/config/index.ts:68]: 20, per slack discussion https://pocket.slack.com/archives/C03E28D1GUD/p1657671668758909

File: src/server/aws/sqs.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import { SQS } from '@aws-sdk/client-sqs';
+import config from '../../config';
+
+export const sqs = new SQS({
+  region: config.aws.region,
+  endpoint: config.aws.endpoint,
+  maxAttempts: 3,
```
Comment (kschelonka) [inline src/server/aws/sqs.ts:7]: Should this be 2? Or should the other one be fixed to 2

**Review by kschelonka — COMMENTED** [review]
lgtm! a couple updates required to fix some copy-paste errors and align batch size/poll rate with slack discussion

File: src/config/index.ts (hunk 4 above)
Comment (sri-kirsh) [reply src/config/index.ts]: #266 updates this

File: src/server/aws/batchDeleteHandler.ts [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+        db: {
+          writeClient: writeClient(),
+          readClient: readClient(),
+        },
+        apiId: 'service', // unused but required for inheritance
+        isPremium: false, //setting default `false` - but it shouldn't matter for delete
+      })
+        .deleteByAnnotationIds(body.annotationIds)
+        .then(() => successCallback('Annotations', userId, traceId));
+    } catch (error) {
```
Comment (sri-kirsh) [inline src/server/aws/batchDeleteHandler.ts:103]: only change in this file is call successCallback on success, and failCallBack on err.

annotationsIds are printed within failCallBack

**Review by kschelonka — APPROVED** [review]

**Pull request merged**